Source/WebCore/platform/graphics/filters/FilterImageTargetSwitcher.cpp

```diff
@@ -41,7 +41,13 @@ FilterImageTargetSwitcher::FilterImageTargetSwitcher(GraphicsContext& destinatio
     if (sourceImageRect.isEmpty())
         return;
 
+#if USE(CAIRO)
+    UNUSED_PARAM(colorSpace);
```

Comment:
I think it's unclear that that colorSpace parameter is now ignored for Cairo. We always enforce SRGB for the m_sourceImage, and auto-convert to linearRGB. While this is entirely correct for SVG filter processing -- it is perfectly fine to create a `FilterImageTargetSwitcher` operating entirely in sRGB, without doing any conversions on non-Cairo platforms. That's now no longer possible.

To avoid this different treatment between the Cocoa/glib ports, I suggest to store the `colorSpace` as `m_colorSpace` member in the class, and use it `endDrawSourceImage` to transform to `m_colorSpace` instead of hardcoding `DestinationColorSpace::LinearSRGB()`.

The net result of the patch stays identical, since SVG filter code passes LinearSRGB as colorSpace, it just removes a port discrepancy and surprises in the future if we ever add treatment for another color space.

Reply:
You are right, and that could be the reason of fails in case of css3/filters/* LayoutTests. Thanks Nikolas!!!